### DISCUSSION TITLE
KDE  After Standby the Network connection fails

### DISCUSSION THREAD

**Discussion by newbie992** [SpiralLinux/SpiralLinux-project#134] [Q&A]
What can be done about this ?
After starting my PC from the Standby the Internet connection ist lost. Nothing helps.
I have to restart the PC in order to get Internet-connection.

I have MX21-KDE and Manjaro-KDE and Endeavour-KDE installed on my PC. 
In all of them this connection problem does not happen.
I hope it can be fixed in SpiralLinux because I like the speed of Spiral . It is faster than the other OS.
Thanks.

**Comment by geckolinux** [comment]
Hi there, sorry for the trouble.

What kind of internet connection is it? WiFI? Cable?

Could you please test this vanilla Debian version to see if it behaves the same?
https://cdimage.debian.org/images/unofficial/non-free/images-including-firmware/current-live/amd64/iso-hybrid/debian-live-11.5.0-amd64-kde+nonfree.iso

Thanks.

**Reply by newbie992** [reply]
Thanks for your answer.
I hate WLAN, therefore I only use Ethernet cable because of EMF radiation.

I will test the version you posted and give feedback.

**Reply by geckolinux** [reply]
Thanks, I'll await your feedback.

**Reply by newbie992** [reply]
First try. I copied the iso on USB-Stick where Ventoy is installed.
Booting from Ventoy usually works without problems, but not with your iso.
Ends up in error messages like:  invalid magic number
and more messages which disappear too fast because the boot-menu of the iso appears.

Now I install your iso on a USB-stick with Balena-etcher.
Sometimes that helps.
I'll give you feedback about this new try.

**Reply by geckolinux** [reply]
OK thanks for letting me know, I removed Ventoy from the list of recommended tools for flashing the USB.

**Reply by newbie992** [reply]
Balena-etcher did the job.
The installed Debian has no problems with Standby.  Booting from Standby works, network finds the Internet and connects fast.
Everything fine as it should be.

I have a question about swappiness.
On this blog they recommend to go from 60 to 25  for PCs which have 4GB RAM or lower.
https://easylinuxtipsproject.blogspot.com/p/speed-mint.html#ID1.1

My PC has 4GB RAM.
Would you advise to do that ?
The blog talks about Linux Mint, not Debian, but maybe it applies in Debian too ?

**Reply by geckolinux** [reply]
> Balena-etcher did the job.
> The installed Debian has no problems with Standby. Booting from Standby works, network finds the Internet and connects fast. Everything fine as it should be.

OK, thanks a lot for that helpful test. Could you please try flashing SpiralLinux to a USB with Balena Etcher and install it with that? If it works well with Debian then SpiralLinux should behave the same. Maybe the previous installation from a Ventoy disk caused the problem.

> I have a question about swappiness.
> On this blog they recommend to go from 60 to 25 for PCs which have 4GB RAM or lower.
> https://easylinuxtipsproject.blogspot.com/p/speed-mint.html#ID1.1
> 
> My PC has 4GB RAM.

It's worth testing. Also note that SpiralLinux uses zRAM swap by default and also includes a more aggressive out-of-memory killer, which might eliminate a lot of the problems that traditional Linux installations used to have in low RAM situations.

**Reply by newbie992** [reply]
I flashed SpiralLinux (KDE) to USB with Balena Etcher and installed it.
Sorry to say that the connection problem is there. After booting from Standby there is no way to get Internet connection.

Maybe I should test SpiralLinux Cinnamon or Mate or Xfce ?

I forgot to mention that I use ext4 formatting only.  In the past I trief btfrs and it ruined everything on my PC (have no idea why, but it looks like btrfs doesn't work in a multi-boot-situation with ext4).
Since SpiralLinux Calamares installer has btrfs as default, I cannot choose "replace" but I need to manually create the ext4 partition for SpiralLinux. And then there is no way to get an uefi-boot-entry.
Thus the booting after installation goes to Manjaro Linux, where I do  "sudo update-grub" which finds the new installation of Spiral and makes the entry in the grub-menu. Then I can boot SpiralLinux from that and it works.
Maybe that is the reason why Standby loses internet connection ?
I don't know.

**Reply by geckolinux** [reply]
Thanks again for testing and reporting.

Have you tried booting both the 5.10 kernel and the 5.18 kernel?

If the problem still happens with both kernels then the SpiralLinux Cinnamon version would be a helpful test to understand if it's a KDE problem or a low-level driver problem.

Could you please post the exact version of the ethernet card you have?

When you say that there is no internet, do you mean that the network device is no longer recognized? Does it receive an IP address? What happens if you open a terminal and run `ping 8.8.8.8` ?

> Since SpiralLinux Calamares installer has btrfs as default, I cannot choose "replace" but I need to manually create the ext4 partition for SpiralLinux. And then there is no way to get an uefi-boot-entry.

During the custom partitioning phase please select the EFI partition and set it to be mounted as `/boot/efi` with the `boot` flag checked.

**Reply by newbie992** [reply]
I have tested many hours now. It is too much to report the details.
The installation of  SpiralLinux KDE and Xfce works, they don't install the uefi-entry correctly.
Both lose the network connection after Standby.  
It is a pity because they are so fast.

I don't have the necessary energy to test another day for hours. 
I go back to Manjaro KDE, MX21 KDE, Linux Mint and Endeavour KDE. They are not that fast but they work without hassles.

Thank you for your patience and help.

**Reply by geckolinux** [reply]
I understand, thanks for your time and effort.

**Reply by newbie992** [reply]
Some feedback.

1. I found the latest Ventoy version on the Ventoy-Website.
    With that version of Ventoy, the installation of SpiralLinux KDE and Xfce is perfect. ---->  Ventoy can be recommended

2. SpiralLinux works on the Laptop of my daughter which has BIOS with MBR. 
    After Standby the internet-connection is there as it must be.   On my PC which has UEFI the Standby loses the internet-connection.
    I installed Peppermint OS , which is based on Debian, on my PC and Standby works perfectly.

I will have a look for new releases of your SpiralLinux. Maybe in the future the problem on UEFI is solved.
BTW, including the sda1 boot partition (uefi) does not work because Calamaris expects at least 300MB of this fat32 uefi-boot partition, but Win10 creates only 100MG and for the multi-boot situation Win10 must be installed first, then the Linux distros.

Strange behavior of Calamares because the distros I mentioned ( Manjaro, Endeavour etc ) accept the 100MB uefi-boot-partition.

**Reply by geckolinux** [reply]
Hi there,

> Have you tried booting both the 5.10 kernel and the 5.18 kernel?

> If the problem still happens with both kernels then the SpiralLinux Cinnamon version would be a helpful test to understand if it's a KDE problem or a low-level driver problem.

> Could you please post the exact version of the ethernet card you have?

> When you say that there is no internet, do you mean that the network device is no longer recognized? Does it receive an IP address? What happens if you open a terminal and run `ping 8.8.8.8` ?

I would need the above information to diagnose what might be going wrong.

> BTW, including the sda1 boot partition (uefi) does not work because Calamaris expects at least 300MB of this fat32 uefi-boot partition, but Win10 creates only 100MG and for the multi-boot situation Win10 must be installed first, then the Linux distros.

In what sense does it not work? Does Calamares throw an error? A screenshot of the custom partitioning screen would be helpful to diagnose what is happening.

**Reply by newbie992** [reply]
`Have you tried booting both the 5.10 kernel and the 5.18 kernel?`
Yes, but not everytime. Most of the time I just used the Live version which appears at the top of the menu.
The results were equally bad. The problem with the uefi-entry and the Standby was not solved by using different kernels.

`If the problem still happens with both kernels then the SpiralLinux Cinnamon version would be a helpful test to understand if it's a KDE problem or a low-level driver problem.`
I already tried KDE and Xfce. I will test Cinnamon asap.

![image](https://user-images.githubusercontent.com/114779823/193469132-8295addf-5b5a-4387-96b5-9852cab5a342.png)

`When you say that there is no internet, do you mean that the network device is no longer recognized? Does it receive an IP address? What happens if you open a terminal and run ping 8.8.8.8 ?`
I don't know. The network cannot connect to the Internet. There are error-messages but I need to make another try and then write them on a paper because right now they are not available.

`In what sense does it not work? Does Calamares throw an error? A screenshot of the custom partitioning screen would be helpful to diagnose what is happening.`
Calamares shows an error-message. It expects 300MB partition fat32 uefi-boot, but I have only 100MB, therefore Calamares is not able to install the grub-entry.
And when I do what you suggested, grabbing the 100MB partition and marking it als uefi-boot   and check the boot, then everything goes wrong, that is the worst thing to do. 
I deleted my comment when that happened because I was so angry. Was not able to boot my PC , ended up in a grub something mess.
Every start of the PC went into that grub thing where no escape other than plugging the power was possible.
I don't remember how I got out of that trap but somehow I did.

How can I make a screenshot while installing ? I have no idea how to manage that. And where to store the screenshot.

**Comment by geckolinux** [comment]
>> Have you tried booting both the 5.10 kernel and the 5.18 kernel?

> Yes, but not everytime.

I would encourage you to try to resolve the standby/resume network issue with the 5.10 kernel, which is the same kernel version that you successfully tested with vanilla Debian. By default SpiralLinux uses the 5.18 kernel, so that could be the source of the problem.

>> If the problem still happens with both kernels then the SpiralLinux Cinnamon version would be a helpful test to understand if it's a KDE problem or a low-level driver problem.
> I already tried KDE and Xfce. I will test Cinnamon asap

Yes, your tests have shown that it is not an issue with KDE or Cinnamon. Rather, it probably an issue with the newer kernel 5.18. Please test it with the 5.10 kernel in any one of the desktop environments that you prefer.

> Calamares shows an error-message. It expects 300MB partition fat32 uefi-boot, but I have only 100MB, therefore Calamares is not able to install the grub-entry.

Actually that is not an error message, it is just a warning that 100MB may be too small, but you can still proceed with the installation. I just now successfully installed SpiralLinux with a 100MB `/boot/efi` partition. However, given the number of different OSes that you have installed that may indeed be too small.

> And when I do what you suggested, grabbing the 100MB partition and marking it als uefi-boot and check the boot, then everything goes wrong, that is the worst thing to do.

That's not exactly what I described, and my instructions were correct. It's not a good idea to do custom partitioning without understanding the concepts. Why not use Btrfs and the automated Replace a partition option?

> How can I make a screenshot while installing ? I have no idea how to manage that. And where to store the screenshot.

You can use the normal screenshot tool that comes with SpiralLinux and then upload it to this forum post directly from the live session.

**Reply by newbie992** [reply]
I installed Cinnamon with kernel 5.10 . During install I left the room for a short time. After returning to the PC, Calamares had disappeared.
No error-messages.

Second try: I installed KDE with kernel 5.10  binding uefi-partiton as your proposed. Installation failed. The error-message is too long, something with uefi  partition.  Taking a screenshot worked, but it is a hassle to find the discussion because there are no bookmarks and I use bitwarden password-manager because I cannot remember 60 passwords for all the forums and websites.
Therefore I did not send a screenshot. 
Makes no sense anyways because the size of the uefi-boot-partition is 100MB ,  93MB already used. There isn't enough space for a new entry, that is why it fails.

So I did the usual procedure which works without defining the uefi-partition. Installation goes through, I boot into Manjaro and do "sudo update-grub",  then restarting the PC  and in the grub-menu  chose SpiralLinux which was fresh installed.
It boots and has internet connection.
Send it to Standby (menu says "Sleep"), then start the PC and SpiralLinux appears, only problem:  internet connection is lost.
There are error-messages but they appear only a few seconds, not enough to make a screenshot.
Something ip-configuration fails and a second error-message "cannot establish connection".
Even when I have the screenshot utility started before the error-messages appear, as soon as I click on the screenshot utility,the error-messages disappear. No chance to make a screenshot.

Since I spent another 3 hours on testing, I give up now.
It is too much hassle. No progress.
I prefer Distros which work . 
Spiral Linux does not work for me.  
Ok, I could decide to not use the Standy option, but I need it often, therefore I stay with that which works.
Sorry for all the time you lost by trying to help me.

**Reply by geckolinux** [reply]
OK, thanks for your time too.

**Reply by newbie992** [reply]
Hello geckolinux,
I am trying again.
Installed Plasma with default btrfs and made sure that I am booting with kernel 5.10 .
The problem is still there.  No internet connection after suspend.
I cannot catch the 2 messages of the network manager, they disappear too fast, but here is the content of the messages:
`ip configuration is not available`
cable connection deactivated
ping 8.8.8.8   ping: connect: Das Netzwerk ist nicht erreichbar``

After restart of the PC the internet connection is there.

I searched a lot with google, but nowhere is a solution which works. Debian forums, arch forums, wiki ubuntuusers.  Nothing which helps.
"sudo systemctl daemon-reload "   doesn't change anything.

Do you have any idea what to do ?

**Reply by geckolinux** [reply]
@newbie992 Please post the output of `sudo dmesg -T` during the time slightly before until slightly after resuming.

**Reply by newbie992** [reply]
`[Sa Nov 12 19:21:31 2022] PM: suspend entry (deep)
[Sa Nov 12 19:21:31 2022] Filesystems sync: 0.054 seconds
[Sa Nov 12 19:21:31 2022] Freezing user space processes ... (elapsed 0.002 seconds) done.
[Sa Nov 12 19:21:31 2022] OOM killer disabled.
[Sa Nov 12 19:21:31 2022] Freezing remaining freezable tasks ... (elapsed 0.001 seconds) done.
[Sa Nov 12 19:21:31 2022] printk: Suspending console(s) (use no_console_suspend to debug)
[Sa Nov 12 19:21:31 2022] parport_pc 00:05: disabled
[Sa Nov 12 19:21:31 2022] sd 4:0:0:0: [sdb] Synchronizing SCSI cache
[Sa Nov 12 19:21:31 2022] sd 0:0:0:0: [sda] Synchronizing SCSI cache
[Sa Nov 12 19:21:31 2022] sd 0:0:0:0: [sda] Stopping disk
[Sa Nov 12 19:21:31 2022] sd 4:0:0:0: [sdb] Stopping disk
[Sa Nov 12 19:21:31 2022] ACPI: Preparing to enter system sleep state S3
[Sa Nov 12 19:21:31 2022] PM: Saving platform NVS memory
[Sa Nov 12 19:21:31 2022] Disabling non-boot CPUs ...
[Sa Nov 12 19:21:31 2022] smpboot: CPU 1 is now offline
[Sa Nov 12 19:21:31 2022] smpboot: CPU 2 is now offline
[Sa Nov 12 19:21:31 2022] smpboot: CPU 3 is now offline
[Sa Nov 12 19:21:31 2022] ACPI: Low-level resume complete
[Sa Nov 12 19:21:31 2022] PM: Restoring platform NVS memory
[Sa Nov 12 19:21:31 2022] Enabling non-boot CPUs ...
[Sa Nov 12 19:21:31 2022] smpboot: Scheduler frequency invariance went wobbly, disabling!
[Sa Nov 12 19:21:31 2022] x86: Booting SMP configuration:
[Sa Nov 12 19:21:31 2022] smpboot: Booting Node 0 Processor 1 APIC 0x2
[Sa Nov 12 19:21:31 2022] CPU1 is up
[Sa Nov 12 19:21:31 2022] smpboot: Booting Node 0 Processor 2 APIC 0x1
[Sa Nov 12 19:21:31 2022] CPU2 is up
[Sa Nov 12 19:21:31 2022] smpboot: Booting Node 0 Processor 3 APIC 0x3
[Sa Nov 12 19:21:31 2022] CPU3 is up
[Sa Nov 12 19:21:31 2022] ACPI: Waking up from system sleep state S3
[Sa Nov 12 19:21:31 2022] usb usb2: root hub lost power or was reset
[Sa Nov 12 19:21:31 2022] usb usb4: root hub lost power or was reset
[Sa Nov 12 19:21:31 2022] sd 0:0:0:0: [sda] Starting disk
[Sa Nov 12 19:21:31 2022] usb usb5: root hub lost power or was reset
[Sa Nov 12 19:21:31 2022] sd 4:0:0:0: [sdb] Starting disk
[Sa Nov 12 19:21:31 2022] parport_pc 00:05: activated
[Sa Nov 12 19:21:32 2022] ata2: SATA link up 1.5 Gbps (SStatus 113 SControl 300)
[Sa Nov 12 19:21:32 2022] ata1: SATA link up 3.0 Gbps (SStatus 123 SControl 300)
[Sa Nov 12 19:21:32 2022] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.SAT0.SPT0._GTF.DSSP], AE_NOT_FOUND (20200925/psargs-330)           
[Sa Nov 12 19:21:32 2022] ACPI Error: Aborting method \_SB.PCI0.SAT0.SPT0._GTF due to previous error (AE_NOT_FOUND) (20200925/psparse-529)              
[Sa Nov 12 19:21:32 2022] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.SAT0.SPT0._GTF.DSSP], AE_NOT_FOUND (20200925/psargs-330)           
[Sa Nov 12 19:21:32 2022] ACPI Error: Aborting method \_SB.PCI0.SAT0.SPT0._GTF due to previous error (AE_NOT_FOUND) (20200925/psparse-529)              
[Sa Nov 12 19:21:32 2022] ata1.00: configured for UDMA/133
[Sa Nov 12 19:21:32 2022] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.SAT0.SPT1._GTF.DSSP], AE_NOT_FOUND (20200925/psargs-330)           
[Sa Nov 12 19:21:32 2022] ACPI Error: Aborting method \_SB.PCI0.SAT0.SPT1._GTF due to previous error (AE_NOT_FOUND) (20200925/psparse-529)              
[Sa Nov 12 19:21:32 2022] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.SAT0.SPT1._GTF.DSSP], AE_NOT_FOUND (20200925/psargs-330)           
[Sa Nov 12 19:21:32 2022] ACPI Error: Aborting method \_SB.PCI0.SAT0.SPT1._GTF due to previous error (AE_NOT_FOUND) (20200925/psparse-529)              
[Sa Nov 12 19:21:32 2022] ata2.00: configured for UDMA/100
[Sa Nov 12 19:21:32 2022] usb 4-2: reset low-speed USB device number 2 using uhci_hcd
[Sa Nov 12 19:21:32 2022] OOM killer enabled.
[Sa Nov 12 19:21:32 2022] Restarting tasks ... 
[Sa Nov 12 19:21:32 2022] pci_bus 0000:04: Allocating resources
[Sa Nov 12 19:21:32 2022] done.
[Sa Nov 12 19:21:32 2022] video LNXVIDEO:00: Restoring backlight state
[Sa Nov 12 19:21:32 2022] PM: suspend exit
[Sa Nov 12 19:21:32 2022] pci 0000:03:00.0: bridge window [mem 0x00100000-0x000fffff 64bit pref] to [bus 04] add_size 200000 add_align 100000
[Sa Nov 12 19:21:32 2022] pci 0000:03:00.0: BAR 15: assigned [mem 0xdfb00000-0xdfcfffff 64bit pref]
[Sa Nov 12 19:21:32 2022] eth0: 0xffffac5180039000, 94:de:80:43:dd:57, IRQ 24
[Sa Nov 12 19:21:35 2022] r8168: eth0: link up
[Sa Nov 12 19:21:35 2022] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[Sa Nov 12 19:21:35 2022] __common_interrupt: 1.34 No irq handler for vector
[Sa Nov 12 19:21:35 2022] ata5: SATA link up 3.0 Gbps (SStatus 123 SControl 300)
[Sa Nov 12 19:21:35 2022] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.SAT0.SPT4._GTF.DSSP], AE_NOT_FOUND (20200925/psargs-330)
[Sa Nov 12 19:21:35 2022] ACPI Error: Aborting method \_SB.PCI0.SAT0.SPT4._GTF due to previous error (AE_NOT_FOUND) (20200925/psparse-529)
[Sa Nov 12 19:21:35 2022] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.SAT0.SPT4._GTF.DSSP], AE_NOT_FOUND (20200925/psargs-330)
[Sa Nov 12 19:21:35 2022] ACPI Error: Aborting method \_SB.PCI0.SAT0.SPT4._GTF due to previous error (AE_NOT_FOUND) (20200925/psparse-529)
[Sa Nov 12 19:21:35 2022] ata5.00: configured for UDMA/133
franz@franz-terrapc:~$ 

   `

**Reply by geckolinux** [reply]
Thanks a lot. Please try uninstalling the `r8168-dkms` package and then reboot.

**Reply by newbie992** [reply]
Thanks, geckolinux.
Amazing results.

Somewhere it says that the r8186-dkms package must be purged. So I did that with Synaptics.

1. Nothing changed with kernel 5.10   No internet after suspend.
2. Success with kernel 5.18   Reboot after suspend shows an  "x" at the network-symbol in panel, but 1 second later the "x" disappears and there is connection.  Internet works.

I uninstalled kernel 5.10 after updating all programs. Synaptic installed kernel 5.19 which is active now.
The problem with internet-connection after reboot from suspend is gone.

I installed SpiralLinux Xfce today. Same problem.
Solution as above:  Purging the r8168-dkms package, then reboot.
Problem gone. Boot from suspend --->  Internet connection works.
Perfect.

Thanks again, geckolinux.
SpiralLinux is faster than all the other distros I tested. And it is stable and responsive with my 4GB RAM PC.
I had distros which are freezing when firefox or chromium plays youtube-videos and thunderbird is started or other programs.
CPU 100% and maybe heavy swapping going on. I had to plug the power to stop the drama.
